src/aead/chacha20_poly1305.rs

```diff
@@ -278,6 +279,67 @@ mod tests {
                     (CRYPTO_POLY1305_STATE_LEN + 255) / 256);
     }
 
+    #[test]
+    pub fn chacha20_tests(){
+        test::from_file("src/chacha_tests.txt", |section, test_case| {
+            assert_eq!(section, "");
+
+            let key_bytes = test_case.consume_bytes("Key");
+            let mut key = [0u32; super::CHACHA20_KEY_LEN/4];
+            for ki in 0..super::CHACHA20_KEY_LEN/4 {
+                let kb = slice_as_array_ref!(&key_bytes[ki * 4..][..4], 4).unwrap();
+                key[ki] = polyfill::slice::u32_from_be_u8(kb);
+            }
+
+            let ctr_bytes = test_case.consume_bytes("Ctr");
+            let mut ctr = [0u32; 4];
+            for ci in 0..4 {
+                let cb = slice_as_array_ref!(&ctr_bytes[ci * 4..][..4], 4).unwrap();
+                ctr[ci] = polyfill::slice::u32_from_be_u8(cb);
+            }
+
+            let input = test_case.consume_bytes("Input");
+            let output = test_case.consume_bytes("Output");
+
+            for len in 0..input.len()+1 {
```

Comment:
So, I don't know what to do style-wise. It is strange that we don't put spaces around `..`` but we do put spaces around `+`, yet `..` binds less tightly than everything. The way I've always resolved this in the code is to use extra parentheses around either end of the range that has a binary operator in it, and then use spaces around binary operators like normal, e.g. `0..(input.len() + 1)`.  I don't know how others feel about this but this seems like the best way to me. I'm open to other ideas, but I'd like to agree on something so that we're using the same style.